File: enigma-machine/python-version/reflector.py
#reflector
class Reflector:
    
    def __init__(self, wiring):
        self.left = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        self.right = wiring
        
    def reflect(self, signal):
        letter = self.right[signal]
        signal = self.left.find(letter)
        return signal
    

File: enigma-machine/javascript-version/reflector.js
//reflector
class Reflector{

    __init__(self, wiring){
        self.left = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        self.right = wiring
    }

    reflect(self, signal){
        letter = self.right[letter]
        signal = self.left.indexOf(letter)
        return letter
    }
}